### PR TITLE
Add title and meta tags

### PR DESCRIPTION
**Pull request by nvolodymyr**

### CODE REVIEW

File: src/app/service/title-meta-tags/title-and-meta-tags.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { TestBed } from '@angular/core/testing';
+import { TitleAndMetaTagsService } from './title-and-meta-tags.service';
+
```
Comment (RuslanHrybchuk) [inline src/app/service/title-meta-tags/title-and-meta-tags.service.spec.ts:3]: unnecessary empty line

File: src/app/service/title-meta-tags/title-and-meta-tags.service.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { Title, Meta } from '@angular/platform-browser';
+
```
Comment (RuslanHrybchuk) [inline src/app/service/title-meta-tags/title-and-meta-tags.service.ts:3]: unnecessary empty line

File: src/app/service/title-meta-tags/title-and-meta-tags.service.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class TitleAndMetaTagsService {
+
+  constructor(
+    private titleService: Title,
+    private meta: Meta
+
```
Comment (RuslanHrybchuk) [inline src/app/service/title-meta-tags/title-and-meta-tags.service.ts:13]: and here too

File: src/app/app.component.ts [hunk 4]
```diff
@@ -26,4 +59,10 @@ export class AppComponent implements OnInit {
       }
     });
   }
+
+  ngOnDestroy(): void {
```
Comment (RomanDeveloperAcc) [inline src/app/app.component.ts:63]: Please delete type, it is not necessary for hooks

File: src/app/service/title-meta-tags/title-and-meta-tags.service.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  ) { }
+
+  initTitleAndMeta(meta): void {
```
Comment (RomanDeveloperAcc) [inline src/app/service/title-meta-tags/title-and-meta-tags.service.ts:16]: Please add access modificator

File: src/app/app.component.ts [hunk 6]
```diff
@@ -26,4 +59,10 @@ export class AppComponent implements OnInit {
       }
     });
   }
+
```
Comment (KimAT25) [inline src/app/app.component.ts:62]: You have all logic in 1 method ngOnInit, please separate all to extra methods and put method in ngOnInit

File: src/app/component/eco-news/eco-news.component.ts [hunk 7]
```diff
@@ -1,18 +1,16 @@
...
-  public getFilterArray: Array<string> = [];
   public actualYear = new Date().getFullYear();
 
-  constructor(private titleService: Title) { }
+  constructor() { }
 
   ngOnInit() {
-    this.titleService.setTitle('Eco News');
   }
+
```
Comment (KimAT25) [inline src/app/component/eco-news/eco-news.component.ts:15]: delete empty line

File: src/app/service/title-meta-tags/title-and-meta-tags.service.spec.ts (hunk 1 above)
Comment (nvolodymyr) [reply src/app/service/title-meta-tags/title-and-meta-tags.service.spec.ts:3]: fixed

File: src/app/service/title-meta-tags/title-and-meta-tags.service.ts (hunk 2 above)
Comment (nvolodymyr) [reply src/app/service/title-meta-tags/title-and-meta-tags.service.ts:3]: +

File: src/app/service/title-meta-tags/title-and-meta-tags.service.ts (hunk 3 above)
Comment (nvolodymyr) [reply src/app/service/title-meta-tags/title-and-meta-tags.service.ts:13]: fix

File: src/app/app.component.ts (hunk 4 above)
Comment (nvolodymyr) [reply src/app/app.component.ts:63]: fixed

File: src/app/service/title-meta-tags/title-and-meta-tags.service.ts (hunk 5 above)
Comment (nvolodymyr) [reply src/app/service/title-meta-tags/title-and-meta-tags.service.ts:16]: fix

File: src/app/component/eco-news/eco-news.component.ts (hunk 7 above)
Comment (nvolodymyr) [reply src/app/component/eco-news/eco-news.component.ts:15]: fixed

File: src/app/app.component.ts (hunk 6 above)
Comment (nvolodymyr) [reply src/app/app.component.ts:41]: fixed

File: src/app/app.component.ts [hunk 8]
```diff
@@ -26,4 +38,37 @@ export class AppComponent implements OnInit {
...
+     .pipe(
+      filter(([title, metas]) => !!(title && metas)),
+      map(([title, metas]) => ({nameMeta: metas[title]})),
+      takeUntil(this.unsubscribe)
+     )
+     .subscribe(({nameMeta }) =>
+      this.titleAndMetaTagsService.initTitleAndMeta(nameMeta));
+ }
+
+  ngOnDestroy() {
```
Comment (nabermi2020) [inline src/app/app.component.ts:70]: Please delete this hook, 'cause you need this subscription during app lifecycle

File: src/app/app.component.ts [hunk 9]
```diff
@@ -26,4 +38,37 @@ export class AppComponent implements OnInit {
...
+}
+  private installTitleAndMetaInBrowser(): void {
+   combineLatest(
+     this.titleSubject,
+     this.metasSubject,
+   )
+     .pipe(
+      filter(([title, metas]) => !!(title && metas)),
+      map(([title, metas]) => ({nameMeta: metas[title]})),
+      takeUntil(this.unsubscribe)
```
Comment (nabermi2020) [inline src/app/app.component.ts:64]: Please delete it

File: src/app/app.component.ts [hunk 10]
```diff
@@ -1,22 +1,34 @@
...
+import { combineLatest, Subject } from 'rxjs';
+import { filter, map, takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent implements OnInit {
+export class AppComponent implements OnInit, OnDestroy  {
```
Comment (nabermi2020) [inline src/app/app.component.ts:14]: Please delete 'OnDestroy'

File: src/app/app.component.ts [hunk 11]
```diff
@@ -26,4 +38,37 @@ export class AppComponent implements OnInit {
       }
     });
   }
+
+  private setMetasSubject(): void {
```
Comment (nabermi2020) [inline src/app/app.component.ts:42]: Please rename it, setter should has some argument

File: src/app/app.component.ts [hunk 12]
```diff
@@ -26,4 +38,37 @@ export class AppComponent implements OnInit {
...
+    this.translations.onDefaultLangChange.subscribe((elem) => {
+    this.metasSubject.next(elem.translations.metas);
+  });
+}
+
+  private setTitleSubject(): void {
```
Comment (nabermi2020) [inline src/app/app.component.ts:48]: the same as line 42

File: src/app/app.component.ts [hunk 13]
```diff
@@ -26,4 +38,37 @@ export class AppComponent implements OnInit {
...
+     this.router.events
+    .pipe(
+     filter((events) => events instanceof NavigationEnd),
+     map((events) => (events as any).url.slice(1)),
+    )
+    .subscribe((nameTitle) => this.titleSubject.next(nameTitle));
+}
+  private installTitleAndMetaInBrowser(): void {
```
Comment (nabermi2020) [inline src/app/app.component.ts:56]: please rename it

File: src/app/app.component.ts [hunk 14]
```diff
@@ -1,22 +1,34 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy  } from '@angular/core';
```
Comment (nabermi2020) [inline src/app/app.component.ts:1]: Delete 'OnDestroy'

File: src/app/app.component.ts [hunk 15]
```diff
@@ -1,22 +1,34 @@
...
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent implements OnInit {
+export class AppComponent implements OnInit, OnDestroy  {
+  private metasSubject = new Subject <string>();
+  private titleSubject = new Subject <any>();
```
Comment (nabermi2020) [inline src/app/app.component.ts:16]: Please specify type

File: src/app/app.component.ts [hunk 16]
```diff
@@ -26,4 +38,37 @@ export class AppComponent implements OnInit {
       }
     });
   }
```
Comment (nabermi2020) [inline src/app/app.component.ts:40]: Please move all functionality below this method to the 'titleAndMetaTagsService' service

File: src/app/app.component.ts (hunk 8 above)
Comment (nvolodymyr) [reply src/app/app.component.ts]: fixed

File: src/app/app.component.ts (hunk 9 above)
Comment (nvolodymyr) [reply src/app/app.component.ts]: fixed

File: src/app/app.component.ts (hunk 10 above)
Comment (nvolodymyr) [reply src/app/app.component.ts]: ok

File: src/app/app.component.ts (hunk 11 above)
Comment (nvolodymyr) [reply src/app/app.component.ts]: ok

File: src/app/app.component.ts (hunk 12 above)
Comment (nvolodymyr) [reply src/app/app.component.ts]: ok

File: src/app/app.component.ts (hunk 13 above)
Comment (nvolodymyr) [reply src/app/app.component.ts]: ok

File: src/app/app.component.ts (hunk 14 above)
Comment (nvolodymyr) [reply src/app/app.component.ts]: ok

File: src/app/app.component.ts (hunk 15 above)
Comment (nvolodymyr) [reply src/app/app.component.ts]: ok

File: src/app/app.component.ts (hunk 16 above)
Comment (nvolodymyr) [reply src/app/app.component.ts]: ok

File: src/app/app.component.ts [hunk 17]
```diff
@@ -1,22 +1,27 @@
...
+export class AppComponent implements OnInit  {
 
   constructor(
     private languageService: LanguageService,
-    private router: Router
+    private titleAndMetaTagsService: TitleAndMetaTagsService,
+    private router: Router,
   ) {}
 
   ngOnInit(): void {
```
Comment (RomanDeveloperAcc) [inline src/app/app.component.ts:19]: Please remove type (it is not necessary for hooks)

Comment (nvolodymyr) [reply src/app/app.component.ts]: fixed

**Pull request merged**